### PR TITLE
Add Amazon Knowledge Bases retriever support

### LINKED ISSUE #3781: Add retriever for Knowledge Bases for Amazon Bedrock
Amazon Bedrock Knowledge Bases is great because it can be a low-cost way to build a domain-specific knowledge base, as it uses OpenSearch serverless, and can ingest and auto-vectorize data from various data sources (e.g. S3) and file formats without any additional work.

https://docs.aws.amazon.com/bedrock/latest/userguide/knowledge-base-api-query.html

https://docs.aws.amazon.com/bedrock/latest/APIReference/API_agent-runtime_Retrieve.html

Marketing page (explainer): https://aws.amazon.com/bedrock/knowledge-bases/

### PR DESCRIPTION
**Pull request by jl4nz**
Fixes #3781

### CODE REVIEW

File: langchain/src/retrievers/amazon_knowledge_bases.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { logVersion010MigrationWarning } from "../util/entrypoint_deprecation.js";
```
Comment (jacoblee93) [inline langchain/src/retrievers/amazon_knowledge_bases.ts:1]: Let's just export from `@langchain/community`, can you revert all changes in `langchain`?

File: libs/langchain-community/src/retrievers/tests/amazon_knowledge_bases.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-process-env */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { test } from "@jest/globals";
+import { AmazonKnowledgeBaseRetriever } from "../amazon_knowledge_bases.js";
+
+test("AmazonKnowledgeBaseRetriever", async () => {
+  const retriever = new AmazonKnowledgeBaseRetriever({
+    topK: 10,
+    knowledgeBaseId: "GZ3S9ZLSIM",
```
Comment (jacoblee93) [inline libs/langchain-community/src/retrievers/tests/amazon_knowledge_bases.int.test.ts:9]: Can you change to an env var?

File: libs/langchain-community/src/retrievers/amazon_knowledge_bases.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      },
+    });
+
+    const retrieveResponse = await this.bedrockAgentRuntimeClient.send(
+      retrieveCommand
+    );
+    const retriveLength = retrieveResponse.retrievalResults?.length;
+    const documents: Array<Document> = [];
+
+    if (retriveLength === 0) {
```
Comment (jacoblee93) [inline libs/langchain-community/src/retrievers/amazon_knowledge_bases.ts:110]: Could just replace this conditional with `.map`?

```ts
return retrieveResponse.retrievalResults?.map((result) => {
        documents.push({
          pageContent: this.cleanResult(result.content?.text || ""),
          metadata: {
            source: result.location?.s3Location?.uri,
            score: result.score,
          },
        });
      }) ?? [];
```

**Comment by jacoblee93** [comment]
Looks good overall, a few nits but should be good to go!

File: langchain/src/retrievers/amazon_knowledge_bases.ts (hunk 1 above)
Comment (jl4nz) [reply langchain/src/retrievers/amazon_knowledge_bases.ts]: Sure, reverted the langchain changes.

File: libs/langchain-community/src/retrievers/tests/amazon_knowledge_bases.int.test.ts (hunk 2 above)
Comment (jl4nz) [reply libs/langchain-community/src/retrievers/tests/amazon_knowledge_bases.int.test.ts]: Changed the knowledgeBaseId and aws region

File: libs/langchain-community/src/retrievers/amazon_knowledge_bases.ts (hunk 3 above)
Comment (jl4nz) [reply libs/langchain-community/src/retrievers/amazon_knowledge_bases.ts]: Done

**Comment by jl4nz** [comment]
@jacoblee93 thanks for the review. Can you have another look with the updates?
Thanks.

**Review by bracesproul — APPROVED** [review]
Thanks for adding this! I pushed up a small commit, just some nits around naming (`bases` -> `base`) and like two other small things. Will get this merged and released tonight!

**Pull request merged**